### ISSUE TITLE
hfst-reweight changes fst behavior, language coverage

### ISSUE THREAD

**Issue by snomos** [hfst/hfst#368]
For [this input file](https://gtsvn.uit.no/langtech/trunk/langs/smn/src/hyphenation/hyphenation.xfscript) I run the following command:

```
printf "save stack hyphenation.hfst\nquit\n" | cat hyphenation.xfscript - | hfst-xfst -p
```

It behaves as follows (notice the comma at the end of the input):

```
$ echo 'mailmist,' | hfst-lookup -q hyphenation.hfst 
mailmist,	mail-mist,	0,000000
```

Then  I add a weight to the fst as follows:

```
hfst-reweight -i hyphenation.hfst -e -a "5000" -o hyphenator-rules-desc-weighted.hfst
```

I test the weighted fst the same way:

```
$ echo 'mailmist,' | hfst-lookup -q hyphenator-rules-desc-weighted.hfst 
mailmist,	mailmist,+?	inf
```

I then remove the comma from the input string:

```
$ echo 'mailmist' | hfst-lookup -q hyphenator-rules-desc-weighted.hfst 
mailmist	mail-mist	5000,000000
```

What is it with reweighting that suddenly makes the comma at the end of the string problematic?

Newest hfst source code as per today, macOS 10.12.6, newest XCode tools.

**Comment by eaxelson** [comment]
Should the transducer 'hyphenation.hfst' recognize any strings that contain a comma in the first place? I think the comma character is not mentioned in file 'hyphenation.xfstscript'. Still, hfst-lookup recognizes 'mailmist,', which is not correct, I guess.

**Comment by snomos** [comment]
The hyphenation fst is a rule transducer, and should accept any input string whatsoever, and output a string with the rules applied. This is what it does before adding a weight, and this is what I expect it to do with a weight. I don't think the comma should have any bearing on the application of the rule fst (except that is expressed in the rules).

This is also how the Xerox tools behave:

```
$ echo 'mailmist,' | lookup -q hyphenation.lookup.xfst 
mailmist,	mail-mist,
```

**Comment by eaxelson** [comment]
This was a bug in hfst-reweight. Fixed in commit 732d645.
